pkg/controller/kfservice/kfservice_controller.go

```diff
@@ -151,6 +151,7 @@ func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Resul
 		knative.NewRouteReconciler(r.Client, r.scheme),
 	}
 
+	kfsvc.Status = v1alpha1.KFServiceStatus{}
```

Comment:
I don't think we want to wipe out all existing statuses. 

This should be cleared up on propagateStatusConditions for the canary. If the canary should be deleted, the propagation should remove it from the status too.

Reply:
done